scripts/build_otp_macos.bash

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+set -euox pipefail
+
+main() {
+  if [ $# -ne 1 ]; then
+    cat <<EOF
+Usage:
+    build_otp_macos.bash ref_name
+EOF
+    exit 1
+  fi
+
+  local ref_name=$1
+  : "${OPENSSL_VERSION:=3.1.6}"
+  : "${OPENSSL_DIR:=/tmp/builds/openssl-${OPENSSL_VERSION}-macos}"
+  : "${WXWIDGETS_VERSION:=3.2.6}"
+  : "${WXWIDGETS_DIR:=/tmp/builds/wxwidgets-${WXWIDGETS_VERSION}-macos}"
+  : "${OTP_DIR:=/tmp/builds/otp-${ref_name}-openssl-${OPENSSL_VERSION}-wxwidgets-${WXWIDGETS_VERSION}-macos}"
+  export MAKEFLAGS=-j$(getconf _NPROCESSORS_ONLN)
+  export CFLAGS="-Os -fno-common -mmacosx-version-min=11.0"
+
+  build_openssl "${OPENSSL_VERSION}"
+  build_wxwidgets "${WXWIDGETS_VERSION}"
+
+  export PATH="${WXWIDGETS_DIR}/bin:$PATH"
+  build_otp "${ref_name}"
+}
+
+build_openssl() {
+  local version=$1
+  local rel_dir="${OPENSSL_DIR}"
+  local src_dir="/tmp/builds/src-openssl-${version}"
+
+  if [ -d "${rel_dir}/bin" ]; then
+    echo "${rel_dir}/bin already exists, skipping build"
+    ${rel_dir}/bin/openssl version
+    return
+  fi
+
+  local ref_name="openssl-${version}"
+  local url="https://github.com/openssl/openssl"
+
+  if [ ! -d ${src_dir} ]; then
+    git clone --depth 1 ${url} --branch ${ref_name} ${src_dir}
```

Comment:
Should this be done in the action so it doesn't have to be done for build macos, windows, ubuntu, etc separately? Could do it once and use `git worktree` to checkout a clean copy to a different directory for each one.

Also had the random thought that maybe this reop should contain a fork so that git tags/releases pointed to something real. The `main` branch would still be as it is but have like `upstream-master` and `upstream-maint` then the nightly script updates those by pulling and pushing from upstream, which causes the build actions to kickoff.

Reply:
I'd prefer to keep as much as possible in the sh script so it can be tested locally. I'd extract stuff out when we add ubuntu. Perhaps instead of build_otp_macos.sh we have build_otp.sh, i.e. it's similar enough across the board that it's easier to have one script than a bunch. WDTT?

Reply:
Regarding making this a fork, brilliant, I think this works. I'll think about it some more but I can't think of a downside, I really like the idea that releases would point to something real (even if forked) and we'd keep it totally unchanged and SHAs will match with upstream.

master/upstream-master is a weird edge case. We need a tag for it so we could have a release so we could store assets for it. The release points to a tag and a SHA and I determined that we can't change the SHA on the release. However (kinda doh) we can remove the tag and create a new one pointing to updated SHA and force push the tag. It seems to work!

WDYT about calling it `master-nightly` and `maint-nightly` instead?

Reply:
Ah yea, `-nightly` works nicely.

Reply:
> maybe this reop should contain a fork

Wouldn't a branch suffice, in this case? As long as you keep the name known. You'd benefit from keeping the logic in the same repo.

I'm not sure I understood the last comment by @wojtekmach, though. Do you prefer a fork or a branch? And what would you see as a downside to having a branch?

Reply:
I'd have to double check it but I vaguely remember that when a repo is a _fork_ GitHub adds some special handling to it, perhaps if you push a commit that @-mentions someone it does _not_ fire a notification from the fork. If there's nothing like that then yeah, doesn't matter if it's a fork or not and I'd keep it as is. But yes we are definitely having all GitHub actions and scripts in this repo on the main branch regardless if it's a fork or not.

Reply:
I've run into issues pushing erlang/otp tags into this repo (they contain .github/workflows changes which need special permissions). I think this is a good idea but I'd like to take it off the table for now, if someone figures it out perhaps we start doing it for OTP 28 or something. In the meantime I'm addicting release notes to our faux releases that point to the upstream ones, see e.g.: https://github.com/erlef/otp_builds/releases/tag/OTP-26.2.5

Reply:
If we figure this out down the road I think we could simply overwrite tags, e.g. `OTP-27.1` instead of pointing to https://github.com/erlef/otp_builds/commit/b5893a3c3a8d0ab54be5d04de450b24d9e5aa149 like all our releases will point to https://github.com/erlang/otp/commit/9ae2ef54fc6dacb5bfbe2dc6fc6f2522d57ce1fb.

I'll resolve this conversation for now.